### PR TITLE
telecom updates for various subscription types

### PR DESCRIPTION
**Pull request by jbranson000**
Includes additional telecom field groupings for general subscription details, as well as television, internet, & landline subscription specifics.

### CODE REVIEW

File: components/mixins/profile/profile-internet-subscription.schema.json [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Internet Subscription",
+  "type": "object",
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:tags": { "industry": ["telecom"] },
+  "meta:intendedToExtend": [
+    "https://ns.adobe.com/xdm/mixins/profile/profile-telecom-subscription"
+  ],
+  "description": "Collects all specific details related to a customer's internet subscription plan, including data cap, connection type, and speed details.",
```
Comment (hanessian) [inline components/mixins/profile/profile-internet-subscription.schema.json:18]: Can we shorten the description where possible? Sorry I know its seems nit picky, but we just updated all XDM descriptions to be as short as possible so I'm trying to keep it consistent. Thanks!

"details related to an internet subscription plan, such as including data cap, connection type, and speed."

File: components/mixins/profile/profile-internet-subscription.schema.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
+{
+  "meta:license": [
+    "Copyright 2020 Adobe Systems Incorporated. All rights reserved.",
+    "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/mixins/profile/profile-internet-subscription",
```
Comment (hanessian) [inline components/mixins/profile/profile-internet-subscription.schema.json:8]: Did verizon set up their subscription schemas as profile, or product? I could see either working, and then we would just need a subscription ID in the profile.

File: components/mixins/profile/profile-landline-subscription.schema.json [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Landline Subscription",
+  "type": "object",
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:tags": { "industry": ["telecom"] },
+  "meta:intendedToExtend": [
+    "https://ns.adobe.com/xdm/mixins/profile/profile-telecom-subscription"
+  ],
+  "description": "Collects all generic details related to a customer's landline subscription plan, including selected features, minutes, and dialing plans.",
```
Comment (hanessian) [inline components/mixins/profile/profile-landline-subscription.schema.json:18]: "landline subscription plan details, including..."

File: components/mixins/profile/profile-landline-subscription.schema.json [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  },
+  "allOf": [
+    {
+      "$ref": "#/definitions/landline-subscription"
+    },
+    {
+      "$ref": "https://ns.adobe.com/xdm/mixins/profile/profile-telecom-subscription"
+    }
+  ],
+  "meta:status": "experimental"
```
Comment (hanessian) [inline components/mixins/profile/profile-landline-subscription.schema.json:88]: same q on this re whether it should be profile or part of the product class.

File: components/mixins/profile/profile-telecom-subscription.schema.json [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Telecom Subscription",
+  "type": "object",
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:tags": {
+    "industry": ["telecom"]
+  },
+  "meta:intendedToExtend": ["https://ns.adobe.com/xdm/context/profile"],
+  "description": "Collects all generic details related to a customer's telecom subscription plan, including pricing, package, and included features.",
```
Comment (hanessian) [inline components/mixins/profile/profile-telecom-subscription.schema.json:18]: "telecom subscription plan details including..."

File: components/mixins/profile/profile-internet-subscription.schema.json [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Internet Subscription",
+  "type": "object",
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:tags": { "industry": ["telecom"] },
+  "meta:intendedToExtend": [
+    "https://ns.adobe.com/xdm/mixins/profile/profile-telecom-subscription"
```
Comment (mjasrotia) [inline components/mixins/profile/profile-internet-subscription.schema.json:16]: I am guessing this is still a mixin which extends profile class and needs to have all the properties from "profile/profile-telecom-subscription". If my assumption is right, please change the "meta:intendedToExtend" to profile class and add a "meta:extends" to add the "profile/profile-telecom-subscription" schema. This applies to all the subscription schemas you are creating in this PR.

File: components/mixins/profile/profile-internet-subscription.schema.json [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+          "properties": {
+            "xdm:dataCap": {
+              "title": "Data Cap",
+              "type": "integer",
+              "description": "Captures the data cap limit for the account, in megabytes (MB)."
+            },
+            "xdm:connectionType": {
+              "title": "Connection Type",
+              "type": "string",
+              "meta:enum": {
```
Comment (mjasrotia) [inline components/mixins/profile/profile-internet-subscription.schema.json:33]: enum missing

File: components/mixins/profile/profile-mobile-plan-details.schema.json [hunk 8]
```diff
@@ -12,7 +12,9 @@
   "meta:extensible": true,
   "meta:abstract": true,
   "meta:tags": { "industry": ["telecom"] },
-  "meta:intendedToExtend": ["https://ns.adobe.com/xdm/context/profile"],
+  "meta:intendedToExtend": [
+    "https://ns.adobe.com/xdm/mixins/profile/profile-telecom-subscription"
```
Comment (mjasrotia) [inline components/mixins/profile/profile-mobile-plan-details.schema.json:16]: it should stay as profile, Add meta:extends to add properties from profile-telecom-subscription

File: components/mixins/profile/profile-mobile-plan-details.schema.json [hunk 9]
```diff
@@ -112,6 +114,31 @@
                     "title": "Ported Number",
                     "type": "boolean",
                     "description": "Captures if the customer ports their number from another carrier."
+                  },
+                  "xdm:devices/5GAccess": {
```
Comment (mjasrotia) [inline components/mixins/profile/profile-mobile-plan-details.schema.json:118]: We do not support / in the property names unless its part of the namespace which is not the case here. Please fix all

**Review by mjasrotia — APPROVED** [review]

**Review by prabhum2 — APPROVED** [review]

**Pull request merged**